File: Question 0227 - Basic Calculator II/Solution.java
import java.util.Stack;

class Solution {
    public int calculate(String s) {
        Stack<Integer> st = new Stack<>();
        int number = 0;
        char op = '+';

        for (int i = 0; i < s.length(); i++) {
            char val = s.charAt(i);
            if (Character.isDigit(val)) {
                number = (number * 10) + (val - '0');
            }

            if ((!Character.isDigit(val) && val != ' ') || (i == s.length() - 1)) {
                if (op == '+') {
                    st.push(number);
                } else if (op == '-') {
                    st.push(-1 * number);
                } else if (op == '*') {
                    st.push(number * st.pop());
                } else if (op == '/') {
                    st.push(st.pop() / number);
                }

                number = 0;
                op = val;
            }
        }

        int ans = 0;
        while(!st.isEmpty()) {
            ans += st.pop();
        }

        return ans;
    }
}

File: Question 0227 - Basic Calculator II/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int calculate(string s)
    {
        stack<int> st;
        char op = '+';
        int number = 0;

        for (int i = 0; i < s.size(); i++)
        {
            char val = s[i];
            if (isdigit(val))
                number = (number * 10) + (val - '0');

            if ((!isdigit(val) && val != ' ') || (i == s.size() - 1))
            {
                if (op == '+')
                    st.push(number);
                else if (op == '-')
                    st.push(-number);
                else if (op == '*')
                {
                    int top = st.top();
                    st.pop();
                    st.push(number * top);
                }
                else if (op == '/')
                {
                    int top = st.top();
                    st.pop();
                    st.push(top / number);
                }

                number = 0;
                op = val;
            }
        }

        int ans = 0;
        while (!st.empty())
        {
            int top = st.top();
            st.pop();
            ans += top;
        }

        return ans;
    }
};